crates/uv/src/commands/pip_show.rs

```diff
@@ -0,0 +1,143 @@
+
+use crate::commands::ExitStatus;
+use crate::printer::Printer;
+
+/// Show information about one or more installed packages.
+pub(crate) fn pip_show(
+    mut packages: Vec<PackageName>,
+    strict: bool,
+    python: Option<&str>,
+    system: bool,
+    quiet: bool,
+    cache: &Cache,
+    mut printer: Printer,
+) -> Result<ExitStatus> {
+    if packages.is_empty() {
+        #[allow(clippy::print_stderr)]
+        {
+            eprintln!(
+                "{}{} Please provide a package name or names.",
+                "warning".yellow().bold(),
+                ":".bold(),
+            );
+        }
+        return Ok(ExitStatus::Failure);
+    }
+
+    // Detect the current Python interpreter.
+    let platform = Platform::current()?;
+    let venv = if let Some(python) = python {
+        PythonEnvironment::from_requested_python(python, &platform, cache)?
+    } else if system {
+        PythonEnvironment::from_default_python(&platform, cache)?
+    } else {
+        match PythonEnvironment::from_virtualenv(platform.clone(), cache) {
+            Ok(venv) => venv,
+            Err(uv_interpreter::Error::VenvNotFound) => {
+                PythonEnvironment::from_default_python(&platform, cache)?
+            }
+            Err(err) => return Err(err.into()),
+        }
+    };
+
+    debug!(
+        "Using Python {} environment at {}",
+        venv.interpreter().python_version(),
+        venv.python_executable().simplified_display().cyan()
+    );
+
+    // Build the installed index.
+    let site_packages = SitePackages::from_executable(&venv)?;
+
+    // Sort and deduplicate the packages, which are keyed by name.
+    packages.sort_unstable();
+    packages.dedup();
+
+    // Map to the local distributions.
+    let distributions = {
+        let mut distributions = Vec::with_capacity(packages.len());
+
+        // Identify all packages that are installed.
+        for package in &packages {
+            let installed = site_packages.get_packages(package);
+            if installed.is_empty() {
+                writeln!(
+                    printer,
+                    "{}{} Package(s) not found for: {}",
+                    "warning".yellow().bold(),
+                    ":".bold(),
+                    package.as_ref().bold()
+                )?;
+            } else {
+                distributions.extend(installed);
+            }
+        }
+
+        distributions
+    };
+
+    // Like `pip`, if no packages were found, return a failure.
+    if distributions.is_empty() {
+        return Ok(ExitStatus::Failure);
+    }
+
+    if !quiet {
+        // Print the information for each package.
+        let mut first = true;
+        for distribution in &distributions {
+            if first {
+                first = false;
+            } else {
+                // Print a separator between packages.
+                #[allow(clippy::print_stdout)]
+                {
+                    println!("---");
```

Comment:
I also had to change these back to using `println`, because the `printer` always prints to `stderr`, but I think it's important that this info goes to `stdout`, so we have to do it "manually" (by passing in `quiet`, etc.). Something we could try to improve in the future.

Reply:
I improved the situation a bit here: https://github.com/astral-sh/uv/pull/2227